scripts/deploy.js

```diff
@@ -138,10 +138,53 @@ async function deployAll(
     };
 }
 
+async function deployAllWithTestGatewayCaller(
```

Comment:
Instead of adding a duplicate, change `deployAll` to accept an optional arg (say `contractOverrides`) to override contracts to use instead of deploying it's own. Example shown below